src/render/md.jl

```diff
@@ -1,17 +1,43 @@
 ## Docs-specific rendering ––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––
 
-function writemime(io::IO, mime::MIME"text/md", docs::Docs{:md})
+function writemime(io::IO, mime::MIME"text/md", docs::Docs{:md}, mdstyle::Dict{ASCIIString, ASCIIString} = DEFAULT_MDSTYLE)
     println(io, docs.data)
 end
 
 ## General markdown rendering ------------------------–––––––––––––––––––––––––––––––––––
 
-function save(file::String, mime::MIME"text/md", doc::Metadata; mathjax = false, include_internal = true)
+const DEFAULT_MDSTYLE = Dict{ASCIIString, ASCIIString}([
+  ("header"         , "#"),
+  ("objname"        , "####"),
+  ("meta"           , "**"),
+  ("exported"       , "##"),
+  ("internal"       , "##"),
+])
```

Comment:
For a case such as this, probably only that it's just a little shorter to type :) Also, the options given to Docile's `@document` and `@docstrings` are  keyed with symbols (technically, they're in a macro so not noticeable externally) so this would give slightly more uniformity across the packages.

I'm not too well-versed on the internals to say whether there's any difference in performance. Interestingly the following tests suggests there might be:

```julia
julia> function test(n,m)
           a = Dict{Symbol, UTF8String}([(:a, n)])
           b = Dict{ASCIIString, UTF8String}([("a", n)])

           @time a[:a]
           @time b["a"]

           @time for i = 1:m
               a[:a]
           end
           @time for i = 1:m
               b["a"]
           end
       end
```

```
julia> test("a", 10000)
elapsed time: 1.052e-6 seconds (0 bytes allocated)
elapsed time: 1.453e-6 seconds (0 bytes allocated)
elapsed time: 0.000403047 seconds (0 bytes allocated)
elapsed time: 0.001128158 seconds (0 bytes allocated)
```